core/src/main/scala/com/eharmony/spotz/Preamble.scala

```diff
@@ -0,0 +1,39 @@
+package com.eharmony.spotz
+
+/**
+  * Definitions that define the default implementation of a point of hyper parameter values.
+  *
+  * @author vsuthichai
+  */
+object Preamble {
+  import scala.language.implicitConversions
+
+  implicit object PointLossOrdering extends Ordering[(Point, Double)] {
+    override def compare(x: (Point, Double), y: (Point, Double)): Int = {
+      if (x._2 > y._2) 1
+      else if (x._2 < y._2) -1
+      else 0
+    }
+  }
+
+  class Point(val hyperParamMap: Map[String, _]) extends Serializable {
```

Comment:
If you make `Point` a `case class` you don't need `Serializable`, `toString`, or `equals`.